src/picohash.h

```diff
@@ -729,6 +729,7 @@ inline void picohash_init_hmac(picohash_ctx_t *ctx, void (*initf)(picohash_ctx_t
         /* hash the key if it is too long */
         picohash_update(ctx, key, key_len);
         picohash_final(ctx, ctx->_hmac.key);
+        ctx->_hmac.hash_reset = ctx->_reset;
         ctx->_hmac.hash_reset(ctx);
```

Comment:
Good catch, that's a nasty bug. I think the mistake is that it computes a hash here not a hmac, so it should actually call `picohash_reset`, no need to set `ctx->_hmac.hash_reset`.
```suggestion
        picohash_reset(ctx);
```

Reply:
This works too